src/android/java/com/pspdfkit/cordova/PSPDFKitCordovaPlugin.java

```diff
@@ -215,8 +220,14 @@
 
     private void showDocument(@NonNull Uri documentUri, @NonNull final PdfActivityConfiguration configuration,
                               @NonNull final CallbackContext callbackContext) {
-        showDocumentForUri(documentUri, configuration);
-        callbackContext.success();
+        if(secret == null){
```

Comment:
Instead of accessing the password as a field, please add another parameter `@Nullable final String password` to the `showDocument()` method, passing in the password that way.